### ISSUE TITLE
Cannot create volume on local-path-provisioner v0.0.17

### ISSUE THREAD

**Issue by sdil** [rancher/local-path-provisioner#147]
Hi, this morning I'm creating a fresh minikube cluster for development. I'm installing local-path-provisioner from [master branch](https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml). And then I'm installing MySQL (using k8s deployment) and RabbitMQ (using Helm).

However, those 2 pods are stuck at pending status for

```
$ kubectl get pods
NAME                                 READY   STATUS    RESTARTS   AGE
cdn-net-ott-mysql-75cc464859-9bp7q   0/1     Pending   0          6m49s
ott-rabbitmq-0                       0/1     Pending   0          6m31s
```

And then I checked pods in all namespaces I see  that create-pvc-* pods are in error state ( I ran this after recreating cluster again).

```
fadhil@fadhil-ThinkPad-T470p ~ $ kubectl get pods --all-namespaces
NAMESPACE            NAME                                              READY   STATUS    RESTARTS   AGE
default              cdn-net-ott-mysql-75cc464859-jxw8s                0/1     Pending   0          26s
default              ott-rabbitmq-0                                    0/1     Pending   0          10s
flux                 helm-operator-76b575f8c8-tbrr4                    1/1     Running   0          27s
kube-system          coredns-6955765f44-hh2c6                          1/1     Running   0          33s
kube-system          coredns-6955765f44-trwxw                          1/1     Running   0          33s
kube-system          etcd-fadhil-thinkpad-t470p                        1/1     Running   0          37s
kube-system          kube-apiserver-fadhil-thinkpad-t470p              1/1     Running   0          37s
kube-system          kube-controller-manager-fadhil-thinkpad-t470p     1/1     Running   0          37s
kube-system          kube-proxy-gz99j                                  1/1     Running   0          32s
kube-system          kube-scheduler-fadhil-thinkpad-t470p              1/1     Running   0          37s
kube-system          storage-provisioner                               1/1     Running   0          40s
local-path-storage   create-pvc-5c2c40fb-d53e-4a27-8e5f-2cf3a0d36b9f   0/1     Error     0          10s
local-path-storage   create-pvc-d5193a35-ef2c-474e-a7ec-1e7861f9fa08   0/1     Error     0          25s
local-path-storage   local-path-provisioner-59d7b946f5-xgmdl           1/1     Running   0          33s
```

From the logs i see as the following:

```

fadhil@fadhil-ThinkPad-T470p ~ $ kubectl logs -n local-path-storage create-pvc-95c23519-0f0e-42a3-a53d-09d7a9ad9a2e
BusyBox v1.32.0 (2020-07-27 19:18:59 UTC) multi-call binary.

Usage: mkdir [OPTIONS] DIRECTORY...

Create DIRECTORY

	-m MODE	Mode
	-p	No error if exists; make parent directories as needed

fadhil@fadhil-ThinkPad-T470p ~ $ kubectl logs -n local-path-storage local-path-provisioner-59d7b946f5-wnpc4
time="2020-10-16T06:40:09Z" level=debug msg="Applied config: {\"nodePathMap\":[{\"node\":\"DEFAULT_PATH_FOR_NON_LISTED_NODES\",\"paths\":[\"/opt/local-path-provisioner\"]}]}" 
time="2020-10-16T06:40:09Z" level=debug msg="Provisioner started" 
I1016 06:40:09.503253       1 leaderelection.go:242] attempting to acquire leader lease  local-path-storage/rancher.io-local-path...
I1016 06:40:09.516424       1 leaderelection.go:252] successfully acquired lease local-path-storage/rancher.io-local-path
I1016 06:40:09.521503       1 event.go:281] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"local-path-storage", Name:"rancher.io-local-path", UID:"ee4b3594-70c5-49e8-ba35-e1397d6d3471", APIVersion:"v1", ResourceVersion:"528", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' local-path-provisioner-59d7b946f5-wnpc4_b6d82446-0d35-4c6e-8831-70c945dec95c became leader
I1016 06:40:09.521571       1 controller.go:773] Starting provisioner controller rancher.io/local-path_local-path-provisioner-59d7b946f5-wnpc4_b6d82446-0d35-4c6e-8831-70c945dec95c!
I1016 06:40:09.621720       1 controller.go:822] Started provisioner controller rancher.io/local-path_local-path-provisioner-59d7b946f5-wnpc4_b6d82446-0d35-4c6e-8831-70c945dec95c!
I1016 06:40:09.621761       1 controller.go:1202] provision "default/mysql-pv-claim" class "local-path": started
time="2020-10-16T06:40:09Z" level=debug msg="config doesn't contain node fadhil-thinkpad-t470p, use DEFAULT_PATH_FOR_NON_LISTED_NODES instead" 
time="2020-10-16T06:40:09Z" level=info msg="Creating volume pvc-95c23519-0f0e-42a3-a53d-09d7a9ad9a2e at fadhil-thinkpad-t470p:/opt/local-path-provisioner/pvc-95c23519-0f0e-42a3-a53d-09d7a9ad9a2e_default_mysql-pv-claim" 
time="2020-10-16T06:40:09Z" level=info msg="create the helper pod create-pvc-95c23519-0f0e-42a3-a53d-09d7a9ad9a2e into local-path-storage" 
I1016 06:40:09.626410       1 event.go:281] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"mysql-pv-claim", UID:"95c23519-0f0e-42a3-a53d-09d7a9ad9a2e", APIVersion:"v1", ResourceVersion:"485", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "default/mysql-pv-claim"
I1016 06:40:22.028331       1 controller.go:1202] provision "default/data-ott-rabbitmq-0" class "local-path": started
time="2020-10-16T06:40:22Z" level=debug msg="config doesn't contain node fadhil-thinkpad-t470p, use DEFAULT_PATH_FOR_NON_LISTED_NODES instead" 
time="2020-10-16T06:40:22Z" level=info msg="Creating volume pvc-d4c92a35-8ec8-4207-bb67-3e45d0c677ee at fadhil-thinkpad-t470p:/opt/local-path-provisioner/pvc-d4c92a35-8ec8-4207-bb67-3e45d0c677ee_default_data-ott-rabbitmq-0" 
time="2020-10-16T06:40:22Z" level=info msg="create the helper pod create-pvc-d4c92a35-8ec8-4207-bb67-3e45d0c677ee into local-path-storage" 
I1016 06:40:22.031439       1 event.go:281] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"data-ott-rabbitmq-0", UID:"d4c92a35-8ec8-4207-bb67-3e45d0c677ee", APIVersion:"v1", ResourceVersion:"618", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "default/data-ott-rabbitmq-0"

fadhil@fadhil-ThinkPad-T470p ~ $ ls /opt/local-path-provisioner/pvc-95c23519-0f0e-42a3-a53d-09d7a9ad9a2e_default_mysql-pv-claim
ls: cannot access '/opt/local-path-provisioner/pvc-95c23519-0f0e-42a3-a53d-09d7a9ad9a2e_default_mysql-pv-claim': No such file or directory
```

I'm using the local-path-provisioner v0.0.16 now and it works fine.

Do you have any idea why local-path-provisoin v0.0.17 is not working on my machine?

I'm using minikube v1.7.2 on Ubuntu 19.10.

**Comment by x-Xymos** [comment]
@sdil how were you able to get it working with the older v0.0.16 version, I changed the image of the deployment but I'm having the same problem.

**Comment by llndqvst** [comment]
I got this issue as well when I tried to install local-path-provisioner today. Had a working install from like 2-3 weeks ago, so that one should have been v0.0.17.

I tried changing the version to v0.0.16 but that didn't fix the problem. What I did find however is that it seems like the create-pvc-... pod is pulling the latest image of busybox. I suspect that something in the official busybox image changed to make it incompatible with local-path-provisoiner.

Description of create-pvc pod:
`Name:         create-pvc-a1bbdc54-97c2-4207-a479-42df0a84b325
Namespace:    local-path-storage
Priority:     0
Node:         k8s-control-1/10.1.2.13
Start Time:   Fri, 16 Oct 2020 15:40:40 +0200
Labels:       <none>
Annotations:  cni.projectcalico.org/podIP: 
              cni.projectcalico.org/podIPs: 
Status:       Failed
IP:           10.244.206.217
IPs:
  IP:  10.244.206.217
Containers:
  local-path-create:
    Container ID:  cri-o://3b5a79c170c5776a582216ba19770cfba5b8a3ae07f7aba3ee0b62d9049b0e3e
    Image:         busybox
    Image ID:      docker.io/library/busybox@sha256:a9286defaba7b3a519d585ba0e37d0b2cbee74ebfe590960b0b1d6a5e97d1e1d
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      /script/setup
      /data/pvc-a1bbdc54-97c2-4207-a479-42df0a84b325_default_local-path-pvc
    State:          Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Fri, 16 Oct 2020 15:40:41 +0200
      Finished:     Fri, 16 Oct 2020 15:40:41 +0200
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /data/ from data (rw)
      /script from script (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-cklhg (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  data:
    Type:          HostPath (bare host directory volume)
    Path:          /opt/local-path-provisioner
    HostPathType:  DirectoryOrCreate
  script:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      local-path-config
    Optional:  false
  default-token-cklhg:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-cklhg
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     
Events:
  Type    Reason   Age   From                    Message
  ----    ------   ----  ----                    -------
  Normal  Pulled   73s   kubelet, k8s-control-1  Container image "busybox" already present on machine
  Normal  Created  73s   kubelet, k8s-control-1  Created container local-path-create
  Normal  Started  73s   kubelet, k8s-control-1  Started container local-path-create`

manifest of busybox:latest
`{
    "manifests": [
        {
            "digest": "sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 527
        },
        {
            "digest": "sha256:a7c572c26ca470b3148d6c1e48ad3db90708a2769fdf836aa44d74b83190496d",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "arm",
                "os": "linux",
                "variant": "v5"
            },
            "size": 527
        },
        {
            "digest": "sha256:ce800872092c37c5f20ef111a5a69c5c8e94d0c5e055f76f530cb5e78a26ec03",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "arm",
                "os": "linux",
                "variant": "v6"
            },
            "size": 527
        },
        {
            "digest": "sha256:6655df04a3df853b029a5fac8836035ac4fab117800c9a6c4b69341bb5306c3d",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "arm",
                "os": "linux",
                "variant": "v7"
            },
            "size": 527
        },
        {
            "digest": "sha256:b8946184ce3ad6b4a09ebad2d85e81cfcaadc6897bfae2e9c6e2a4fe6afa6ee0",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "arm64",
                "os": "linux",
                "variant": "v8"
            },
            "size": 527
        },
        {
            "digest": "sha256:ba65e8d39e89b5c16f036c88c85952756777bf5385bce148bc44be48fac37d94",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "386",
                "os": "linux"
            },
            "size": 527
        },
        {
            "digest": "sha256:d7e83316d74e150866d82c45de342e78f662fe0aefbdb822d7d10c8b8e39cc4b",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "mips64le",
                "os": "linux"
            },
            "size": 527
        },
        {
            "digest": "sha256:0a11a95568b680dce6906a015bed88381e28ad17b31a63f7fec057b35573235a",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "ppc64le",
                "os": "linux"
            },
            "size": 528
        },
        {
            "digest": "sha256:426c855775f026d3fe76988b71938f4c9dc6840f09c0f29d8d4c75cc4238503b",
            "mediaType": "application\/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "s390x",
                "os": "linux"
            },
            "size": 528
        }
    ],
    "mediaType": "application\/vnd.docker.distribution.manifest.list.v2+json",
    "schemaVersion": 2
}`

The busybox image seems to have changed 2 days ago, so it might be worth testing with a image older than that.

**Comment by mlbiam** [comment]
Same issue here.  Fresh install on kubeadm 1.19

**Comment by mlbiam** [comment]
i don't think this is a busybox issue.  `absolutePath` isn't being set.  It doesn't look like `OPTARGS` is being passed to the helper:

```
kubectl logs create-pvc-e22d5e60-b20c-4899-bd5e-e510e869e176 -n local-path-storage
KUBERNETES_SERVICE_PORT=443
KUBERNETES_PORT=tcp://10.96.0.1:443
HOSTNAME=create-pvc-e22d5e60-b20c-4899-bd5e-e510e869e176
SHLVL=1
HOME=/root
KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
KUBERNETES_PORT_443_TCP_PORT=443
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443
KUBERNETES_SERVICE_HOST=10.96.0.1
PWD=/
BusyBox v1.32.0 (2020-10-12 23:47:18 UTC) multi-call binary.

Usage: mkdir [OPTIONS] DIRECTORY...

Create DIRECTORY

	-m MODE	Mode
	-p	No error if exists; make parent directories as needed
```

**Comment by mlbiam** [comment]
confirming, reverting to 16 makes it work

**Comment by llndqvst** [comment]
> confirming, reverting to 16 makes it work

Huh, that's weird. Downgrading to v0.0.16 does nothing on my end..

**Comment by yasker** [comment]
@llndqvst @mlbiam @sdil @x-Xymos Sorry for the trouble. I've released v0.0.18 and the chart should work correctly now.

**Comment by sdil** [comment]
Thank you @yasker for the fix! :smile: 

I'm closing this issue now.
